test/models/PostModel/test.js

```diff
@@ -0,0 +1,267 @@
+      const { post: receivedPost } = await PostModel.getOne(
+        _.pick(post, 'author', 'permlink'),
+      );
+      expect(receivedPost).to.be.deep.eq(_.omit(post, 'post_id'));
+    });
+    it('Should check post for identity using root_author key ', async () => {
+      const { post: receivedPost } = await PostModel.getOne(
+        _.pick(post, 'root_author', 'permlink'),
+      );
+      expect(receivedPost).to.be.deep.eq(_.omit(post, 'post_id'));
+    });
+    it('Should return null post when request name cannot be found', async () => {
+      const { post: receivedPost } = await PostModel.getOne(
+        { author: faker.name.firstName() },
+      );
+      expect(receivedPost).to.be.null;
+    });
+    it('Should check that the error exists', async () => {
+      sinon.stub(Post, 'findOne').throws('DataBase is not responding');
+      const { error } = await PostModel.getOne({ author: faker.name.firstName() });
+      expect(error).to.be.exist;
+    });
+  });
+  describe('On findByBothAuthors', async () => {
+    let post, nameAuthor, userPermlink, rootAuthor, countPosts;
+    beforeEach(async () => {
+      countPosts = faker.random.number(30);
+      nameAuthor = faker.name.firstName();
+      userPermlink = faker.random.string(10);
+      rootAuthor = { author: nameAuthor, permlink: userPermlink };
+      for (let i = 0; i < countPosts; i++) {
+        if (i === 0) {
+          post = await PostFactory.Create(
+            { author: nameAuthor, permlink: userPermlink, rootAuthor },
+          );
+        } else if (i === 1) await PostFactory.Create({ rootAuthor });
+        else await PostFactory.Create();
+      }
+    });
+    afterEach(async () => {
+      sinon.restore();
+    });
+    it('Should return the requested post if the parameters match', async () => {
+      const { result: [result] } = await PostModel.findByBothAuthors(
+        { author: nameAuthor, permlink: userPermlink },
+      );
+      expect(result).to.be.deep.eq(_.omit(post, 'post_id'));
+    });
+    it('Should return empty array when request params cannot be found', async () => {
+      const { result } = await PostModel.findByBothAuthors(
+        { author: faker.name.findName(), permlink: faker.random.string(10) },
+      );
+      expect(result).to.be.empty;
+    });
+    it('Should check that the error exists', async () => {
+      sinon.stub(Post, 'find').throws('DataBase is not responding');
+      const { error } = await PostModel.findByBothAuthors(
+        { author: faker.name.findName(), permlink: faker.random.string(10) },
+      );
+      expect(error).to.be.exist;
+    });
+  });
+  describe('On getBlog', async () => {
+    let author, permLink, permLink2, latestPosts, countWobj, countPosts, latestPost, wObjects;
+    beforeEach(async () => {
+      countWobj = faker.random.number(10);
+      countPosts = faker.random.number(10);
+      latestPosts = new Mongoose.Types.ObjectId();
```

Comment:
latest posts must be an array! rewrite it correctly, then i check the cases themselves